Kentor.AuthServices/IdentityProvider.cs

```diff
@@ -130,6 +130,31 @@ public Saml2BindingType Binding
             }
         }
 
+        private IPendingAuthStorageContainer pendingAuthStorageContainer;
+
+        /// <summary>
+        /// How pending authentication metadata is stored before sending Authentication Requests 
+        /// to the identity provider and retrieved again after a response is returned. Defaults
+        /// to an in-memory container. 
+        /// </summary>
+        public IPendingAuthStorageContainer PendingAuthStorageContainer

```

Comment:
I don't think that this belongs on the IdentityProvider level. It should probably be part of ISPOptions, or maybe IOptions.